playbooks/roles/freeipa_server/tasks/install_freeipa_client.yml

```diff
@@ -0,0 +1,50 @@
+---
+- name: 'Add /etc/hosts record for FreeIPA'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  lineinfile: dest=/etc/hosts regexp=".*{{ ansible_nodename }}$" line="{{ freeipa_server_ip }} {{ ansible_nodename }}" state=present
+
+- name: 'Install FreeIPA client and admintools'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  yum: name=ipa-client,ipa-admintools state=latest
+
+- name: 'Register Foreman in FreeIPA'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  shell: ipa-client-install --domain example.com --server {{ ansible_nodename }} --mkhomedir --fixed-primary --realm={{ freeipa_server_realm }} --force-join -p {{ freeipa_server_kerberos_principal }}@{{ freeipa_server_realm }} -w {{ freeipa_server_realm_password }} -U creates=/etc/krb5.keytab
+
+- name: 'Run foreman-prepare-realm'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  shell: echo changeme | foreman-prepare-realm {{ freeipa_server_kerberos_principal }}@{{ freeipa_server_realm }} {{ katello_server }}.{{ freeipa_server_domain }} chdir=/root creates=/root/freeipa.keytab
+
+- name: 'Create HTTP principal'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  shell: echo changeme | kinit {{ freeipa_server_kerberos_principal }}
+
+- name: 'Add the host'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  shell: ipa host-add --ip-address {{ ansible_eth0.ipv4.address }} {{ katello_server }}.{{ freeipa_server_domain }}
+
+- name: 'Add the service'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  shell: ipa service-add HTTP/{{ katello_server }}.{{ freeipa_server_domain }} --force
+
+- name: 'Get the keytab'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  shell: ipa-getkeytab -s {{ ansible_nodename }} -p HTTP/{{ katello_server }}.{{ freeipa_server_domain }}@{{ freeipa_server_realm }} -k /etc/httpd/conf/ipa.keytab
+
+- name: 'Copy the keytab'
+  become: true
+  delegate_to: "{{ katello_server }}"
+  shell: cp /root/freeipa.keytab /etc/foreman-proxy/freeipa.keytab && chown foreman-proxy /etc/foreman-proxy/freeipa.keytab && chmod 0600 /etc/foreman-proxy/freeipa.keytab
```

Comment:
Oh, it's confusing but it's correct - there's 2 keytabs, the one on line 40 is the HTTP principal for authentication as a user.  

This line https://github.com/Katello/forklift/pull/263/files/dee320284b8ab64fb692406ec33291403f0965ec#diff-2abddccb9cad83fddc0669f2cc87cb81R20 generates /etc/foreman-proxy/freeipa.keytab for use by the proxy.

@akofink It might be clearer if you rearrange the order so this is with foreman prepare realm, or change the name so it's clearer which keytab this is.